dashboard/src/interfaces/tabs/ProtocolTab.tsx

```diff
@@ -123,6 +124,45 @@ function ProtocolTab(
         });
       };
 
+      list[entityName] = {};
+      for (let x = 0; x < Object.keys(entitiesData[entityName]).length; x++) {
+        const entityField = Object.keys(entitiesData[entityName])[x];
+        if (entityField === 'timestamp') {
+          continue
+        }
+        const renderedField = dataFields[entityField];
+        if (renderedField) {
+          list[entityName][entityField] = "Included";
+        } else {
+          const extrapolatedFields = Object.keys(dataFields).filter((df: string) => {
+            return df.includes(entityField);
+          });
+          if (extrapolatedFields?.length > 0) {
+            list[entityName][entityField] = "Array Included";
+          } else {
+            const req = '!' === entitiesData[entityName][entityField].split("")[entitiesData[entityName][entityField].split("").length - 1]
```

Comment:
is this just checking if the text has an `!`?

Reply:
This checks whether or not that entity field is listed as required in the schema or not. I did it this way to avoid the redundancy of embedding another level of key/value pairs to have this property as a boolean. The text being checked is a field type followed by an exclamation point if required. Sometimes the field type specified on the schema has to be checked to determine some conditions about how that field should be handled.

Reply:
The reason I bring it up is that it looks like a good use case for some regex.
`!!/!/g.exec(entitiesData[entityName][entityField])`
Should return true/false if there is an `!` in the string

Reply:
Not critical but it was hard to read what it was doing so I wanted to bring it up